javascript/javascript1/week3/cactusIO.js

```diff
@@ -0,0 +1,114 @@
+// Adding an activity
+
+let activities = [];
+const usageLimit = 60; //Usage limit is 60 minutes
+var today = new Date();
+
+function addActivity(date, activity, duration) {
+  if (typeof date !== "string") {
+    console.log("the date needs to be a string!");
+    process.exit();
+  }
+  if (typeof activity !== "string") {
+    console.log("the activity needs to be a string!");
+    process.exit();
+  }
+  if (typeof duration !== "number") {
+    console.log("the duration needs to be a number!");
+    process.exit();
+  }
+  if (date === "today") {
+    date = today.toISOString().slice(0, 10);
+  }
+
+  activityObject = {
+    date: date,
+    activity: activity,
+    duration: duration,
+  };
```

Comment:
There is a little javascript trick you can use here.

If the key (left of colons) and the value (right of colon) is the exact same, you do not have to repeate yourself. 

```suggestion
  activityObject = {
    date,
    activity,
    duration,
  };
```

try it out! Remember this only works when you the key and value is the same. for example `date: date`